code/AssetLib/FBX/FBXConverter.cpp

```diff
@@ -1451,34 +1425,38 @@ unsigned int FBXConverter::ConvertMeshMultiMaterial(const MeshGeometry &mesh, co
 }
 
 static void copyBoneToSkeletonBone(aiMesh *mesh, aiBone *bone, aiSkeletonBone *skeletonBone ) {
-    skeletonBone->mNumnWeights = bone->mNumWeights;
-    skeletonBone->mWeights = bone->mWeights;
+    skeletonBone->mNumWeights = bone->mNumWeights;
+    skeletonBone->mWeights = new aiVertexWeight[skeletonBone->mNumWeights];
+    ::memcpy(skeletonBone->mWeights, bone->mWeights, skeletonBone->mNumWeights * sizeof(aiVertexWeight));
     skeletonBone->mOffsetMatrix = bone->mOffsetMatrix;
     skeletonBone->mMeshId = mesh;
     skeletonBone->mNode = bone->mNode;
     skeletonBone->mParent = -1;
 }
 
-void FBXConverter::ConvertWeightsToSkeleton(aiMesh *out, const MeshGeometry &geo, const aiMatrix4x4 &absolute_transform, aiNode *parent, unsigned int materialIndex,
-        std::vector<unsigned int> *outputVertStartIndices, SkeletonBoneContainer &skeletonContainer) {
-
-    if (skeletonContainer.SkeletonBoneToMeshLookup.find(out) != skeletonContainer.SkeletonBoneToMeshLookup.end()) {
-        return;
-    }
+void FBXConverter::FinalizeSkeletons() {
+    std::vector<bool> skins_set(mSkeletons.size(), false);
+    for (unsigned int i = 0; i < mMeshes.size(); ++i) {
+        aiMesh *mesh = mMeshes[i];
 
-    ConvertWeights(out, geo, absolute_transform, parent, materialIndex, outputVertStartIndices);
-    skeletonContainer.MeshArray.emplace_back(out);
-    SkeletonBoneArray *ba = new SkeletonBoneArray;
-    for (size_t i = 0; i < out->mNumBones; ++i) {
-        aiBone *bone = out->mBones[i];
-        if (bone == nullptr) {
+        const unsigned int skinId = mesh2skin_map[i];
+        if (skinId == AI_NO_SKELETONS || skins_set[skinId])
             continue;
+
+        aiSkeleton *skeleton = mSkeletons[skinId];
+        skeleton->mNumBones = mesh->mNumBones;
+        skeleton->mBones = new aiSkeletonBone*[skeleton->mNumBones]();
+        for (size_t j = 0; j < mesh->mNumBones; ++j) {
+            aiBone *bone = mesh->mBones[j];
```

Comment:
to avoid junk data in the pointer, should the skeleton's bone pointer be set to null?
such as: 
skeleton->mBones[j] = nullptr